actionpack/test/controller/render_test.rb

```diff
@@ -162,6 +162,17 @@ def conditional_hello_with_cache_control_headers
     render action: "hello_world"
   end
 
+  def conditional_hello_with_expires_and_confliciting_cache_control_headers
+    response.headers["Cache-Control"] = "no-cache, must-revalidate"
+    expires_now
```

Comment:
This triggers the edge case that matters. I ran into this in an actual app by a controller calling `send_data`. I had a controller action trying to manually set a very conservative cache header of `no-cache, no-store, must-revalidate`. But, because send_data [modifies the cache control hash](https://github.com/rails/rails/blob/master/actionpack/lib/action_controller/metal/data_streaming.rb#L149), The normalization logic in `set_conditional_cache_control` kicked in and the `must-revalidate` was stripped. That seems fine, but it was weird because all the other actions in this controller were sending the full `no-cache, no-store, must-revalidate` with no problem. But, this was only be happenstance, since none of the other actions had some bit of rails implicitly modify the `cache_control` hash.